src/backend/services/metrics.py

```diff
@@ -383,6 +402,32 @@ def initialize_sdk_metrics_data(
     )
 
 
+def initialize_sdk_metrics_data(
```

Comment:
Why do we need another function for rerank?

Reply:
I wanted to make some changes to the chat without affecting rerank; we currently do not have any tests for rerank so the risks are higher

Reply:
Thanks for clarifying!

Reply:
re: https://github.com/cohere-ai/cohere-toolkit/pull/416#discussion_r1677566109

Do you know why it is currently not available? Does it make sense to log this metric if we cannot know the tokens in/out?

I was also thinking: we do not need signals for ever chat event. Is there a way to remove the hooks into the async stream with decorators? It can simplify the logic. All we really need atm is:

- chat successful: sum up tokens, report model, report agent, report success
- chat failed/interrupted: sum up tokens, report model, report agent, report failed
happy to discuss offline